apps/buildog/src/app/(blog-pages)/blog/page.tsx

```diff
@@ -1,85 +1,167 @@
 "use client";
 
-import React, { useEffect, useState } from "react";
+import React, { useState } from "react";
 import { Button } from "@ui/components/button";
 import { DataTable } from "@ui/components/ui/data-table";
 import {
   DropdownMenu,
   DropdownMenuContent,
   DropdownMenuItem,
-  DropdownMenuLabel,
   DropdownMenuSeparator,
   DropdownMenuTrigger,
 } from "@ui/components/dropdown-menu";
 
+import { Badge } from "@ui/components/ui/badge";
+
+import { CalendarIcon, Pencil2Icon, CheckCircledIcon } from "@ui/components/ui/react-icons";
+
+// Define the Blog type
 type Blog = {
   cards: {
     id: string;
     value: string;
     lock: boolean;
   }[];
+  title: string;
   filename: string;
   text: string;
+  status: string;
+  tags: string[];
+  slug?: string;
 };
 
-type Blogs = {
-  [filename: string]: Blog;
-};
+// Define the Blogs type as an array of Blog objects
+type Blogs = Blog[];
```

Comment:
You don't have to define this type. You can directly give Blog[] to useState type